Healthy/Classes/ReusableViews/FormTextField/FormTextField.swift

```diff
@@ -0,0 +1,33 @@
+
+import UIKit
+
+class FormTextField: UIView {
+    
+    // MARK: - Outlets
+    //
+    
+    @IBOutlet weak private(set) var contentView: UIView!
+    @IBOutlet weak private(set) var titleLabel: UILabel!
+    @IBOutlet weak private(set) var inputTextField: UITextField!
+    @IBOutlet weak private(set) var errorLabel: UILabel!
+    
+    
+    // MARK: - init
+    //
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        initView()
+    }
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        initView()
+    }
+    
+    private func initView() {
+        Bundle.main.loadNibNamed(FormTextFieldResources.nibName, owner: self, options: nil)
```

Comment:
You can use this handy extension to load nibs
``` swift
extension UIView {
    class func loadNib<T: UIView>() -> T {
      return Bundle(for: T.self).loadNibNamed(String(describing: T.self), owner: nil, options: nil)!.first as! T
    }
}
````

and your code will be like this 
``` swift

let view: UIView = .loadNib()

``` 

:warning: Just be careful, for this extension to work, you **must** name the XIB file exactly like the view class name.

Also an improvement for this code, you can create a factory to create the view for you.

``` swift
class FormTextFieldFactory {
  func createView(title: String) -> FormTextField {
    var view: FormTextField = .loadNib
    view.setTitle(title) // just an example
    return view
  }
}

let factory = FormTextFieldFactory()
factory.createView(title: "Awesome") // FormTextField view
```

Reply:
okay, i will read more about factory and apply on this view.
but i have a question! if i don't using this view by programmatically just using by outlet in this way factory is useful ???

Reply:
Factory pattern makes creating objects easier for you, you pass the changing values to the factory and it hides from you all the mess like loading from nib in our case.
Check this amazing resource  https://refactoring.guru/design-patterns/factory-comparison

Reply:
It's against the approach of using nib files in the project. The factory would be great if we're following programmatic views.